File: Medium/128. Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        int count=0;
        int max=0;
        if(nums.size()==0)
        {
            return 0;
        }
        set<int> uset(nums.begin(), nums.end());
        auto y = uset.begin();

        for(auto x: uset)
        {
            max = max>count?max:count;
            y++;
            if(x==*(y)-1){
                count++;
            }
            else 
            {
                count=0;
            } 
            
        }
        return max+1;
    }
};

File: Medium/128. Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> hash_set = new HashSet<Integer>();
        Arrays.sort(nums);
        for(int i: nums)
        {
            hash_set.add(i);
        }
        int maxCount = 0;
        for(int i=0;i<nums.length;i++)
        {
            int count = 1;
            int temp = nums[i];
            if(!hash_set.contains(temp-1))
            {
                while(hash_set.contains(temp+1))
                {
                    count++;
                    temp++;
                }
            }
            maxCount = Math.max(maxCount,count);
        }
        return maxCount;
    }
}
